nixos/modules/services/monitoring/adguard-exporter.nix

```diff
@@ -0,0 +1,118 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let cfg = config.services.adguard-exporter;
+in {
+  options.services.adguard-exporter = {
+    enable = mkEnableOption "adguard-exporter";
+
+    interval = mkOption {
+      type = types.str;
+      default = "10s";
+      description = ''
+        Interval of time the exporter will fetch data from Adguard
+      '';
+    };
+
+    protocol = mkOption {
+      type = types.str;
+      default = "http";
+      description = ''
+        Protocol to use to query Adguard (i.e. the http scheme, either http or https)
+      '';
+    };
+
+    hostname = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
```

Comment:
`localhost` is probably a better default since that also includes `::1` as a target.